packages/labs/src/tests/frameworks/react/create-component_test.ts

```diff
@@ -0,0 +1,214 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {UpdatingElement} from 'updating-element';
+import {property} from 'updating-element/decorators/property.js';
+import {customElement} from 'updating-element/decorators/customElement.js';
+import 'react/umd/react.development.js';
+import 'react-dom/umd/react-dom.development.js';
+import {createComponent} from '../../../frameworks/react/create-component.js';
+import {assert} from '@esm-bundle/chai';
+
+const elementName = 'basic-element';
+@customElement(elementName)
+class BasicElement extends UpdatingElement {
+  @property({type: Boolean})
+  bool = false;
+  @property({type: String})
+  str = '';
+  @property({type: Number})
+  num = -1;
+  @property({type: Object})
+  obj: {[index: string]: unknown} | null | undefined = null;
+  @property({type: Array})
+  arr: unknown[] | null | undefined = null;
+
+  @property({type: Boolean, reflect: true})
+  rbool = false;
+  @property({type: String, reflect: true})
+  rstr = '';
+  @property({type: Number, reflect: true})
+  rnum = -1;
+  @property({type: Object, reflect: true})
+  robj: {[index: string]: unknown} | null | undefined = null;
+  @property({type: Array, reflect: true})
+  rarr: unknown[] | null | undefined = null;
+
+  fire(name: string) {
+    this.dispatchEvent(new Event(name));
+  }
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    [elementName]: BasicElement;
+  }
+}
+
+suite('React createComponent', () => {
+  let container: HTMLElement;
+
+  setup(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  teardown(() => {
+    if (container && container.parentNode) {
+      container.parentNode.removeChild(container);
+    }
+  });
+
+  const BasicElementComponent = createComponent<
```

Comment:
Can you rewrite this as:

```ts
  const basicEvents = {
    onFoo: 'foo',
    onBar: 'bar',
  };
  const BasicElementComponent = createComponent<BasicElement, typeof basicEvents>(
    window.React, elementName, basicEvents);
```

just to show in tests how you can avoid duplicating the events type?

Reply:
Done.